### PR TITLE
Add ClientIP Connection property

### PR DESCRIPTION
**Pull request by ColinSullivan1**
Add the ClientIP connection property to retrieve the server info client IP field.

Signed-off-by: Colin Sullivan <colin@synadia.com>

### CODE REVIEW

**Review by kozlovic — APPROVED** [review]
LGTM

File: src/NATS.Client/Conn.cs [hunk 1]
```diff
@@ -987,6 +988,27 @@ public string ConnectedUrl
...
+        {
+            get
+            {
+                lock (mu)
+                {
+                    if (string.IsNullOrEmpty(info.client_ip))
+                        return null;
+
+                    return IPAddress.Parse(info.client_ip);
+                }
```
Comment (watfordgnf) [inline src/NATS.Client/Conn.cs]: We can avoid locking during the IP address parsing:
```suggestion
                string clientIp;
                lock (mu)
                {
                    clientIp = info.client_ip;
                }
                
                return !String.IsNullOrEmpty(clientIp) ? IPAddress.Parse(clientIp) : null;
```

File: src/Tests/IntegrationTests/TestConnection.cs [hunk 2]
```diff
@@ -571,8 +571,17 @@ public void CanConnectWhenHandshakeTimeoutIsSpecified()
             }
         }
 
-        /// NOT IMPLEMENTED:
-        /// TestErrOnMaxPayloadLimit
+        [Fact]
```
Comment (watfordgnf) [inline src/Tests/IntegrationTests/TestConnection.cs:574]: A category or other attribute describing the minimum NATS version for some of these feature tests would be nice:
```suggestion
        [Fact]
        [Trait("Category", "NATS 2.1.6+")]
```

File: src/NATS.Client/Conn.cs [hunk 3]
```diff
@@ -26,6 +26,7 @@
 using System.Diagnostics;
 using NATS.Client.Extensions;
 using NATS.Client.Internals;
+using System.Net;
```
Comment (watfordgnf) [inline src/NATS.Client/Conn.cs]: `using`s  got out of order.

**Review by watfordgnf — APPROVED** [review]
One comment and two suggestions, but otherwise looks good!

File: src/NATS.Client/Conn.cs (hunk 1 above)
Comment (ColinSullivan1) [reply src/NATS.Client/Conn.cs]: Thanks - good catch.

**Review by kozlovic — APPROVED** [review]
LGTM

**Review by watfordgnf — APPROVED** [review]

**Comment by ColinSullivan1** [comment]
Thanks for the reviews!  Once tests are green, I'll merge.

**Pull request merged**